File: ExperimentsData/Services/Impl/AttributeService.cs
using System;
using System.Collections.Generic;
using AutoMapper;
using ExperimentsData.Models.DAO;
using ExperimentsData.Models.DTO;
using ExperimentsData.Repositories;

namespace ExperimentsData.Services.Impl
{
    public class AttributeService : IAttributeService
    {
        private IAttributeRepository _repository;
        private IDatasetRepository _datasetRepository;
        private ISampleRepository _sampleRepository;
        private IMapper _mapper;

        public AttributeService(ISampleRepository sampleRepository, IDatasetRepository datasetRepository, IAttributeRepository repository, IMapper mapper)
        {
            _sampleRepository = sampleRepository;
            _datasetRepository = datasetRepository;
            _repository = repository;
            _mapper = mapper;
        }

        public List<AttributeRegisterDto> getAll(Guid sampleId)
        {
            List<AttributeEntity> result = _repository.GetAllBySampleId(sampleId);
            return _mapper.Map<List<AttributeRegisterDto>>(result);
        }

        public AttributeRegisterDto create(Guid sampleId, AttributeRegisterDto registerDto)
        {
            var entity = _mapper.Map<AttributeEntity>(registerDto);
            entity.Id = new Guid();
            entity.SampleEntity = _sampleRepository.GetById(sampleId);
            if(entity.SampleEntity == null) throw new ArgumentException("THIS SAMPLE DOES NOT EXISTS.");
            entity.SampleEntityId = entity.SampleEntity.Id;
            _repository.Save(entity);
            return registerDto;
        }
    }
}

File: attributes.module/src/main/java/br/com/lukkascost/attributes/module/services/impl/AttributeService.java
package br.com.lukkascost.attributes.module.services.impl;

import br.com.lukkascost.attributes.module.mappers.AttributeMapper;
import br.com.lukkascost.attributes.module.services.IAttributeService;
import br.com.lukkascost.commons.module.models.dto.AttributeCreateDTO;
import br.com.lukkascost.commons.module.models.dto.AttributeDetailsDTO;
import br.com.lukkascost.commons.module.models.dto.AttributeValueDTO;
import br.com.lukkascost.commons.module.models.entities.AttributeEntity;
import br.com.lukkascost.commons.module.models.entities.SampleEntity;
import br.com.lukkascost.commons.module.repositories.IAttributeRepository;
import br.com.lukkascost.commons.module.repositories.IDatasetRepository;
import br.com.lukkascost.commons.module.repositories.ISampleRepository;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.stereotype.Service;

import java.util.UUID;

@Service
public class AttributeService implements IAttributeService {

    private final IAttributeRepository attributeRepository;
    private final ISampleRepository sampleRepository;
    private final IDatasetRepository datasetRepository;
    private final AttributeMapper attributeMapper;

    public AttributeService(IAttributeRepository attributeRepository, ISampleRepository sampleRepository, IDatasetRepository datasetRepository, AttributeMapper attributeMapper) {
        this.attributeRepository = attributeRepository;
        this.sampleRepository = sampleRepository;
        this.datasetRepository = datasetRepository;
        this.attributeMapper = attributeMapper;
    }

    @Override
    public AttributeDetailsDTO create(AttributeCreateDTO dto) {

        SampleEntity datasetEntity = sampleRepository.findById(dto.getSampleId()).get();
        Specification<AttributeEntity> spec = attributeMapper.convert(dto);
        AttributeEntity entity = attributeRepository.findOne(spec).orElse(null);
        if (entity!= null ) return attributeMapper.convert(entity);
        entity = attributeMapper.convertEntity(dto);
        entity.setId(UUID.randomUUID());
        entity.setSample(datasetEntity);
        entity = attributeRepository.save(entity);
        return attributeMapper.convert(entity);
    }

    @Override
    public Page<AttributeDetailsDTO> findAll(AttributeDetailsDTO attributeDetailsDTO, UUID sample_id, Pageable pageable) {
        Specification<AttributeEntity> spec = attributeMapper.convert(attributeDetailsDTO, sample_id);
        Page<AttributeEntity> entities = attributeRepository.findAll(spec, pageable);
        return attributeMapper.convert(entities);
    }

    @Override
    public AttributeValueDTO findAllValues(AttributeDetailsDTO attributeDetailsDTO, UUID sample_id, Pageable pageable) {
        Specification<AttributeEntity> spec = attributeMapper.convert(attributeDetailsDTO, sample_id);
        Page<AttributeEntity> entities = attributeRepository.findAll(spec, pageable);

        return attributeMapper.convertValue(entities.getContent());
    }
}
